include/deal.II/distributed/solution_transfer.h

```diff
@@ -373,12 +373,27 @@ namespace parallel
       register_data_attach();
     };
 
-
-  } // namespace distributed
+    namespace Legacy
+    {
+      /**
+       * @deprecated Use dealii::parallel::distributed::SolutionTransfer
+       * without the DoFHandlerType template instead.
+       */
+      template <int dim,
+                typename VectorType,
+                typename DoFHandlerType = DoFHandler<dim>>
+      class DEAL_II_DEPRECATED SolutionTransfer
+        : public dealii::parallel::distributed::
+            SolutionTransfer<dim, VectorType, DoFHandlerType>
+      {
+        using dealii::parallel::distributed::
+          SolutionTransfer<dim, VectorType, DoFHandlerType>::SolutionTransfer;
+      };
+    } // namespace Legacy
+  }   // namespace distributed
 } // namespace parallel
```

Comment:
I'd much rather we have just one top-level `namespace Legacy` in which we replicate the namespaces from which we draw.